### ISSUE TITLE
Question about datasets

### ISSUE THREAD

**Issue by rtrahms** [milesial/Pytorch-UNet#243]
Still researching how the DataSet class works, but maybe you know...
I was successful in training using the Carvana dataset, putting images in one folder and masks in another.  It is implied (I think) with that dataset that there must be a 1-1 relationship between image files in one folder and masks in another (with a name modifier like '_mask').  What happens if there are images in the folder that have no associated mask?  Does it skip those, or does it barf with an exception?  The reason I ask is I have a similar set of images and masks that I would like to use for my own training set, but some images in the collection don't have masks, but most do.  I'd rather not have to look through and delete the images that have no masks.

Any guidance here would be helpful.
Thanks,
Rob

**Comment by milesial** [comment]
If will lead to an exception as you can read here https://github.com/milesial/Pytorch-UNet/blob/381924dfd6396c92b17ceebaa9c607f3f9392106/utils/dataset.py#L50-L51

You can just change the assert to an `if` and try the next idx in `self.ids`
